From dd9c8b3e67147ab232415eca00e7e2e033e04010 Mon Sep 17 00:00:00 2001
From: Luca Magrone <luca@magrone.cc>
Date: Mon, 7 Oct 2024 21:45:12 +0200
Subject: [PATCH] Add missing cstring and cstdint headers

This allows to have definitions for uint8_t and memcpy in some source
files under some versions of GCC

Signed-off-by: Luca Magrone <luca@magrone.cc>
---
 cie-pkcs11/LOGGER/Logger.cpp | 1 +
 cie-pkcs11/LOGGER/Logger.h   | 1 +
 cie-pkcs11/Util/log.h        | 1 +
 3 files changed, 3 insertions(+)

diff --git a/cie-pkcs11/LOGGER/Logger.cpp b/cie-pkcs11/LOGGER/Logger.cpp
index 3584444..3df99b5 100644
--- a/cie-pkcs11/LOGGER/Logger.cpp
+++ b/cie-pkcs11/LOGGER/Logger.cpp
@@ -2,6 +2,7 @@
 #include <cstdlib>
 #include <ctime>
 #include <iomanip>
+#include <cstring>
 #include <sys/stat.h>
 #include <unistd.h>
 #include "Logger.h"
diff --git a/cie-pkcs11/LOGGER/Logger.h b/cie-pkcs11/LOGGER/Logger.h
index 5890527..358c30d 100644
--- a/cie-pkcs11/LOGGER/Logger.h
+++ b/cie-pkcs11/LOGGER/Logger.h
@@ -6,6 +6,7 @@
 #include <fstream>
 #include <sstream>
 #include <string>
+#include <cstdint>
 
 #include <errno.h>
 #include <pthread.h>
diff --git a/cie-pkcs11/Util/log.h b/cie-pkcs11/Util/log.h
index 58089f5..9b0a88a 100644
--- a/cie-pkcs11/Util/log.h
+++ b/cie-pkcs11/Util/log.h
@@ -4,6 +4,7 @@
 //#include "defines.h"
 #include <string>
 #include <wintypes.h>
+#include <cstdint>
 
 #ifndef OutputDebugString
 #define OutputDebugString printf
-- 
2.43.5

